src/compiler/checker.ts

```diff
@@ -280,6 +280,11 @@ namespace ts {
         const resolvingSignature = createSignature(undefined, undefined, undefined, emptyArray, anyType, /*typePredicate*/ undefined, 0, /*hasRestParameter*/ false, /*hasLiteralTypes*/ false);
         const silentNeverSignature = createSignature(undefined, undefined, undefined, emptyArray, silentNeverType, /*typePredicate*/ undefined, 0, /*hasRestParameter*/ false, /*hasLiteralTypes*/ false);
 
+        const unresolvedTypePredicate: void & { __unresolvedTypePredicate: void } = (() => {

```

Comment:
I want to ensure that this is definitely *not* useable as a `TypePredicate` -- it's just one to prevent polymorphism. Otherwise it's too easy to access `resolvedTypePredicate` and think you're getting the right thing.